### DISCUSSION TITLE
Create createSelectors for similar functionality to createFeature for deeper properties of state

### DISCUSSION THREAD

**Discussion by david-shortman** [ngrx/platform#3099] [Ideas]
The new `createFeature` functionality handily provides selectors which can be exported by destructuring the feature object.

But, this only provides selectors for top-level fields.

It would add convenience and cut boilerplate if there was a way to generate basic selectors from other selectors.

Suppose `createSelectors` existed like so:

```typescript
export const selectSomething = () => ({ one: 1, two: 2, three: 3 });

export const { selectOne, selectTwo, selectThree } = createSelectors(selectSomething);
```

**Comment by markostanimirovic** [comment]
@david-shortman 

You can use `createChildSelectors` from `ngrx-child-selectors` package: https://github.com/markostanimirovic/ngrx-child-selectors#createchildselectors

**Reply by david-shortman** [reply]
I think that’s pretty cool. Have you considered it being a strong inclusion in core ngrx? It seems more natural now that createFeature is included out of the box.
